clients/drcachesim/tools/invariant_checker.cpp

```diff
@@ -1029,15 +1062,37 @@ invariant_checker_t::parallel_shard_memref(void *shard_data, const memref_t &mem
                      shard->file_type_)) {
             if (type_is_read(memref.data.type)) {
                 // Skip D-filtered traces which don't have every load or store records.
-                report_if_false(shard, shard->expected_read_records_ > 0,
-                                "Too many read records");
+                report_if_false(
+                    shard,
+                    shard->expected_read_records_ >
+                        0 IF_X86(
+                            // iret pops bunch of data from the stack at interrupt
+                            // returns. Read count cannot be determined using the
+                            // decoder. System call trace templates collected on
+                            // QEMU would show all reads.
+                            || (shard->between_kernel_syscall_trace_markers_ &&
+                                shard->prev_instr_.decoding.opcode == OP_iret)
+                            // Xrstor does multiple reads. Read count cannot be
+                            // determined using the decoder. System call trace
```

Comment:
You need a cpuid-queried value to determine the size IIRC.  It ends up being kind of like the SVE vector length: should the trace store the data in the headers?  At least add a comment on how we'd do it.  Simulators may well want that info.

Reply:
Well, the thing is that the cpuid value may be different for the system where the template is generated and where the trace being injected into was collected.

Reply:
Added comment.